ingestr/src/appsflyer/client.py

```diff
@@ -89,13 +90,12 @@ def fetch_campaigns(
             "cohort_day_3_total_revenue_per_user",
             "cohort_day_7_total_revenue_per_user",
             "cohort_day_7_revenue_per_user",
-            "cohort_day_14_total_revenue_per_user",
-            "cohort_day_14_revenue_per_user",
-            "cohort_day_21_total_revenue_per_user",
-            "cohort_day_21_revenue_per_user",
             "retention_day_7",
         ]
-        return self._fetch_data(start_date, end_date, metrics=metrics)
+        max_cohort_duration = 7
+        max_allowed_end_date = (datetime.now() - timedelta(days=max_cohort_duration)).strftime('%Y-%m-%d')
+        adjusted_end_date = min(end_date, max_allowed_end_date)
+        return self._fetch_data(start_date, adjusted_end_date, metrics=metrics)
```

Comment:
why?

Reply:
We can only fetch cohort metric data from more than 7 days ago. If a user tries to fetch cohort data for today, Appsflyer shows an error "429." To handle this, we limit the end date to 7 days before the current date. For example, if a user is trying to fetch data from 2024-09-10 to 2024-10-20, the data will be fetched from 2024-09-10 to 2024-10-13 instead. Otherwise, it will show an error, and we won't be able to fetch data (this issue occurs only in the campaigns resource).